Bugzilla/Net/Google.pm

```diff
@@ -0,0 +1,169 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+#
+# This Source Code Form is "Incompatible With Secondary Licenses", as
+# defined by the Mozilla Public License, v. 2.0.
+
+package Bugzilla::Net::Google;
+
+use 5.10.1;
+use strict;
+use warnings;
+
+use HTTP::Headers;
+use HTTP::Request;
+use List::Util qw(none);
+use Mojo::JSON qw(decode_json);
+use Mojo::JWT;
+use Moo;
+use Try::Tiny;
+use Types::Standard qw(Int Str);
+use URI::Escape     qw(uri_escape_utf8);
+
+use Bugzilla::Error;
+
+extends 'Bugzilla::Net::Storage';
+
+has access_token        => (is => 'rw', isa => Str,);
+has access_token_expiry => (is => 'rw', isa => Int,);
+has service_account     => (is => 'ro', isa => Str,);
+
+my $PATH_BASE = 'storage/v1/b';
+
+##################
+# Public methods #
+##################
+
+# Return type of net storage
+sub data_type { return 'google'; }
+
+###################
+# Private methods #
+###################
+
+# Based on the type of action being performed, select the proper
+# method and the path needed for the Google API.
+sub _get_method_path {
+  my ($self, $action, $key) = @_;
+  my ($method, $format);
+  my @args = ($self->bucket);
+
+  if (none { $action eq $_ } qw(add head get delete)) {
+    ThrowCodeError('net_storage_invalid_action', {action => $action});
+  }
+
+  if ($action eq 'add') {
+    $format
+      = 'upload/'
+      . $PATH_BASE
+      . '/%s/o?uploadType=media&name='
+      . uri_escape_utf8($key);
+    $method = 'POST';
```

Comment:
Can we build the format string here with `@args` in the same way as the other branches of this `if` statement? It's consistent that way, and we can also get rid of all the `push @args, $key;` lines and just have one after the `if` block.